plugins/pulp_docker/plugins/models.py

```diff
@@ -199,14 +191,27 @@ def from_json(cls, manifest_json, digest):
         :param digest:        The content digest of the manifest, as described at
                               https://docs.docker.com/registry/spec/api/#content-digests
         :type  digest:        basestring
+        :param tag:           Tag of the image repository
+        :type  tag:           basestring
+        :param upstream_name: Name of the upstream repository
+        :type  upstream_name: basestring
 
         :return:              An initialized DockerManifest object
         :rtype:               pulp_docker.common.models.DockerManifest
         """
+        # manifest schema2 manifest does not contain tag and name information
+        # we need to retrieve them from other sources, that's why there were added 2 more
+        # parameters in this method
         manifest = json.loads(manifest_json)
-        fs_layers = [FSLayer(blob_sum=layer['blobSum']) for layer in manifest['fsLayers']]
-        return cls(digest=digest, name=manifest['name'], tag=manifest['tag'],
-                   schema_version=manifest['schemaVersion'], fs_layers=fs_layers)
+        config_layer = None
+        try:
+            fs_layers = [FSLayer(blob_sum=layer['digest']) for layer in manifest['layers']]
+            config_layer = manifest['config']['digest']
+        except:
```

Comment:
Can this specifically catch a ``KeyError``? Or at least some narrow list of exceptions?